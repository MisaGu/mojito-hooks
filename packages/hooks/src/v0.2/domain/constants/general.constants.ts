export const config = {
  CONTENTFUL_URL: process.env.CONTENTFUL_URL || '',
  CONTENTFUL_AUTH_TOKEN: process.env.CONTENTFUL_AUTH_TOKEN,
  MOJITO_API_URL: process.env.MOJITO_API_URL || '',
  INVOICE_URL: process.env.INVOICE_URL || '',
  ORGANIZATION_ID: process.env.API_ORGANIZATION_ID || '8fb128bd-f55d-4bcc-8b6c-0beb684e4d4e',
  MARKETPLACE_ID: process.env.MARKETPLACE_ID || 'c5ac4f84-b78b-4cb7-a24f-7f0c38da0eb2',
  TEST_ENV: process.env.TEST_ENV || '',
  AUTH0_DOMAIN: process.env.AUTH0_DOMAIN || '',
  AUTH0_CLIENTID: process.env.AUTH0_CLIENTID || '',
  AUTH_REDIRECT_URI: process.env.AUTH_REDIRECT_URI || '',
};
