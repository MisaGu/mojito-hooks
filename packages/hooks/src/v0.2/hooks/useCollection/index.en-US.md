---
nav:
  path: /hooks
  group: Mojito
---

# useCollection

Returns the selected collection, with lots data when available.

## Examples

<code src="./demo/demo1.tsx" />
