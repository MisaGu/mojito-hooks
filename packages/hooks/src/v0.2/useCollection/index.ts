import { useAuth0 } from '@auth0/auth0-react';
import { config } from '@constants';
import { IMojitoCollection } from '@interfaces';
import { contentfulQueries, EContentfulQueries, EMojitoQueries, mojitoQueries } from '@state';
import { gqlRequest, queryClient } from '@utils';
import { contentfulNormalizer, mojitoNormalizer } from '@utils/gqlDataNormalizer.util';
import { useRouter } from 'next/router';
import { QueryObserverResult, useQuery, UseQueryOptions } from 'react-query';
import { contentfulGqlClient, useContentfulAuctionsSlugList } from './useContentful';
import { mojitoGqlClient, useMarketplaceCollectionsSlugWithItemsId } from './useMojito';

export function useCollection<TSelectorResult = undefined>(
  props: {
    url?: string;
    slug?: string;
    options?: UseQueryOptions<any>;
    selector?: (state: IMojitoCollection) => TSelectorResult;
  } = {},
): {
  slug: string;
  isAuction: boolean;
  isFakeAuction: boolean;
} & (TSelectorResult extends undefined
  ? {
      collection: IMojitoCollection | null;
    } & Omit<QueryObserverResult<IMojitoCollection, unknown>, 'data'>
  : { data?: TSelectorResult }) {
  const { getIdTokenClaims } = useAuth0();
  const { marketplaceCollectionsSlugWithItemsId } = useMarketplaceCollectionsSlugWithItemsId();
  const { auctionsSlugList } = useContentfulAuctionsSlugList();
  const router = useRouter();

  const auctionSlug =
    props?.slug ?? (props?.url ?? router.asPath).split('#')[0].split('?')[0].split('/')[1];
  const collectionByPath = marketplaceCollectionsSlugWithItemsId?.find(
    (e) => e.slug == auctionSlug,
  );

  const isAuction = !!collectionByPath && auctionsSlugList.includes(auctionSlug);
  const isFakeAuction = !!collectionByPath && !auctionsSlugList.includes(auctionSlug);
  const queryKey = [
    `Mojito ${EMojitoQueries[EMojitoQueries.collectionBySlug]}`,
    {
      slug: auctionSlug,
      marketplaceID: config.MARKETPLACE_ID,
    },
  ];

  const { data, ...result } = useQuery(
    queryKey,
    async () => {
      const token = await getIdTokenClaims();
      if (token) {
        mojitoGqlClient.setHeader('authorization', `Bearer ${token.__raw}`);
      }

      if (!isAuction && !isFakeAuction) return null;
      const collectionItems = collectionByPath.items.map((e) => e.id);

      await Promise.all([
        queryClient.prefetchQuery(
          [
            `Contentful ${EContentfulQueries[EContentfulQueries.auctionBySlug]}`,
            { slug: auctionSlug },
          ],
          gqlRequest.bind(null, {
            query: contentfulQueries[EContentfulQueries.auctionBySlug],
            variables: { slug: auctionSlug },
            normalizerFn: contentfulNormalizer,
            gqlClient: contentfulGqlClient,
          }),
        ),
        queryClient.prefetchQuery(
          [`Contentful ${EContentfulQueries[EContentfulQueries.shortLots]}`, { slug: auctionSlug }],
          gqlRequest.bind(null, {
            query: contentfulQueries[EContentfulQueries.shortLots],
            variables: {
              slug: auctionSlug,
              mojitoIds: collectionItems,
            },
            normalizerFn: contentfulNormalizer,
            gqlClient: contentfulGqlClient,
          }),
        ),
      ]);

      return await gqlRequest<IMojitoCollection>({
        query: mojitoQueries[EMojitoQueries.collectionBySlug],
        variables: {
          slug: auctionSlug,
          marketplaceID: config.MARKETPLACE_ID,
        },
        normalizerFn: mojitoNormalizer,
        gqlClient: mojitoGqlClient,
      });
    },
    // @ts-ignore
    props?.options ? { ...props?.options } : {},
  );

  // @ts-ignore
  return {
    slug: isAuction || isFakeAuction ? auctionSlug : '',
    isAuction,
    isFakeAuction,
    ...(props?.selector ? { data } : { collection: data ?? null, ...result }),
  };
}
