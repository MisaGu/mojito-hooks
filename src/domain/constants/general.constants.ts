export const config = {
  CONTENTFUL_URL:
    'https://graphql.contentful.com/content/v1/spaces/fu9did2d8yaw/environments/staging' ||
    process.env.CONTENTFUL_URL ||
    '',
  CONTENTFUL_AUTH_TOKEN:
    '19vUSnF3_8S-OsepxXBcDAI_Ua3GbwSy5c7HNTXB-R0' || process.env.CONTENTFUL_AUTH_TOKEN || '',
  MOJITO_API_URL: 'https://api-dev.mojito.xyz/query' || process.env.MOJITO_API_URL || '',
  INVOICE_URL: 'https://sothebys.dev.mojito.xyz/invoice' || process.env.INVOICE_URL || '',
  MARKETPLACE_ID: 'd15b1b02-3c87-434d-a7fb-004b27415fbc' || process.env.MARKETPLACE_ID || '',
  ORGANIZATION_ID: 'a1dc174d-abf4-47d3-be3c-9281c6eba456' || process.env.ORGANIZATION_ID || '',
  AUTH0_DOMAIN: 'https://accounts.staging.sothelabs.com' || process.env.AUTH0_DOMAIN || '',
  AUTH0_CLIENTID: 'thTFyyXuq94Zr6nBIfEVnQ5iUwRlJsSg' || process.env.AUTH0_CLIENTID || '',
  AUTH_REDIRECT_URI: 'http://localhost:3000' || process.env.AUTH_REDIRECT_URI || '',
};
