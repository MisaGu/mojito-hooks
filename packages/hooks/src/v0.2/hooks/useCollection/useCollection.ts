import { useQueryClient } from 'react-query';
import { config } from '../../domain/constants/general.constants';
import {
  IContentfulAuctionsQuery,
  IMojitoCollection,
  IMojitoMarketplaceResponse,
} from '../../domain/interfaces';
import { getCollectionSlug } from '../../domain/utils/path.util';
import { EMojitoQueries } from '../../domain/gql/queries';
import { EContentfulQueries } from '../../domain/gql/contentful';
import { QueryKey } from '../../domain/utils/queryKeyFactory.util';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';
import { BaseQueryHookPropsWithUrlAndSlug } from '../../domain/interfaces/hooks.interface';

export type UseCollectionData = IMojitoCollection | null | undefined;

export type UseCollectionReturn = ReturnType<typeof useCollection>;

export type UseCollectionProps = BaseQueryHookPropsWithUrlAndSlug<UseCollectionData>;

export function useCollection(props?: UseCollectionProps) {
  const queryClient = useQueryClient();
  const collectionSlug = getCollectionSlug(props);

  async function preloadFn() {
    const [mojitoCollections, contentfulCollectionSlugsOnly] = await Promise.all([
      queryClient.fetchQuery<IMojitoMarketplaceResponse>(
        QueryKey.get(EMojitoQueries.marketplaceCollectionsInfoWithItemsIdAndSlug, {
          id: config.MARKETPLACE_ID,
        }),
      ),
      queryClient.fetchQuery<IContentfulAuctionsQuery>(
        QueryKey.get(EContentfulQueries.auctionsSlugList),
      ),
    ]);

    const mojitoCollection = mojitoCollections.marketplace.collections.find(
      (collection) => collection.slug == collectionSlug,
    );

    const contentfulCollection = contentfulCollectionSlugsOnly.auctionCollection.items.find(
      (collection) => collection.slug == collectionSlug,
    );

    if (!mojitoCollection) return null;

    if (contentfulCollection) {
      const collectionItems = mojitoCollection.items.map((item) => item.id);

      await Promise.all([
        queryClient.prefetchQuery(
          QueryKey.get(EContentfulQueries.auctionBySlug, { slug: collectionSlug }),
        ),
        queryClient.prefetchQuery(
          QueryKey.get(EContentfulQueries.shortLots, { mojitoIds: collectionItems }),
        ),
      ]);
    }
  }

  return useMojitoFactory({
    as: 'collection',
    query: EMojitoQueries.collectionBySlug,
    variables: {
      slug: collectionSlug,
      marketplaceID: config.MARKETPLACE_ID,
    },
    options: props?.options,
    preloadFn,
  });
}
