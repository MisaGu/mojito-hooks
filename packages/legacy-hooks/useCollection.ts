import { useAuth0 } from '@auth0/auth0-react';
import { QueryObserverResult, useQuery, UseQueryOptions } from 'react-query';
import { EMojitoQueries, mojitoQueries } from '../data';
import { contentfulQueries, EContentfulQueries } from '../data/graph_ql/contentful.query';
import { config } from '../domain/general.constants';
import { IMojitoCollection } from '../domain/interfaces';
import { contentfulNormalizer, getPath, gqlRequest, mojitoNormalizer, queryClient } from '../utils';
import { contentfulGqlClient, useContentfulAuctionsSlugList } from './useContentful';
import { mojitoGqlClient, useMarketplaceCollectionsSlugWithItemsId } from './useMojito';

/*

THIS WHOLE FILE HAS BEEN MOVED TO V2.

*/

export function useCollection<TSelectorResult = undefined>(
  props: {
    url?: string;
    slug?: string;
    options?: UseQueryOptions<any>;
    selector?: (state: IMojitoCollection) => TSelectorResult;
  } = {},
): {
  slug: string;
  isAuction: boolean;
  isFakeAuction: boolean;
} & (TSelectorResult extends undefined
  ? {
      collection: IMojitoCollection | null;
    } & Omit<QueryObserverResult<IMojitoCollection, unknown>, 'data'>
  : { data?: TSelectorResult }) {
  const { getIdTokenClaims } = useAuth0();
  const { marketplaceCollectionsSlugWithItemsId } = useMarketplaceCollectionsSlugWithItemsId();
  const { auctionsSlugList } = useContentfulAuctionsSlugList();

  // const marketplaceCollectionsSlugWithItemsId = queryClient.getQueriesData([
  //   `Mojito ${EMojitoQueries[EMojitoQueries.marketplaceCollectionsInfoWithItemsIdAndSlug]}`,
  //   { id: config.MARKETPLACE_ID },
  // ]);

  // const auctionsSlugList = queryClient.getQueriesData([
  //   `Contentful ${EContentfulQueries[EContentfulQueries.auctionsSlugList]}`,
  // ]);

  const auctionSlug = getPath[1];
  const collectionByPath = marketplaceCollectionsSlugWithItemsId?.find(
    (e) => e.slug == auctionSlug,
  );

  const isAuction = !!collectionByPath && auctionsSlugList.includes(auctionSlug);
  const isFakeAuction = !!collectionByPath && !auctionsSlugList.includes(auctionSlug);
  const queryKey = [
    `Mojito ${EMojitoQueries[EMojitoQueries.collectionBySlug]}`,
    {
      slug: auctionSlug,
      marketplaceID: config.MARKETPLACE_ID,
    },
  ];

  const { data, ...result } = useQuery(
    queryKey,
    async () => {
      const token = await getIdTokenClaims();
      if (token) {
        mojitoGqlClient.setHeader('authorization', `Bearer ${token.__raw}`);
      }

      if (!isAuction && !isFakeAuction) return null;
      const collectionItems = collectionByPath?.items?.map((e) => e.id);

      await Promise.all([
        queryClient.prefetchQuery(
          [
            `Contentful ${EContentfulQueries[EContentfulQueries.auctionBySlug]}`,
            { slug: auctionSlug },
          ],
          gqlRequest.bind(null, {
            query: contentfulQueries[EContentfulQueries.auctionBySlug],
            variables: { slug: auctionSlug },
            normalizerFn: contentfulNormalizer,
            gqlClient: contentfulGqlClient,
          }),
        ),
        queryClient.prefetchQuery(
          [`Contentful ${EContentfulQueries[EContentfulQueries.shortLots]}`, { slug: auctionSlug }],
          gqlRequest.bind(null, {
            query: contentfulQueries[EContentfulQueries.shortLots],
            variables: {
              slug: auctionSlug,
              mojitoIds: collectionItems,
            },
            normalizerFn: contentfulNormalizer,
            gqlClient: contentfulGqlClient,
          }),
        ),
      ]);

      return await gqlRequest<IMojitoCollection>({
        query: mojitoQueries[EMojitoQueries.collectionBySlug],
        variables: {
          slug: auctionSlug,
          marketplaceID: config.MARKETPLACE_ID,
        },
        normalizerFn: mojitoNormalizer,
        gqlClient: mojitoGqlClient,
      });
    },
    // @ts-ignore
    props?.options ? { ...props?.options } : {},
  );

  // @ts-ignore
  return {
    slug: isAuction || isFakeAuction ? auctionSlug : '',
    isAuction,
    isFakeAuction,
    ...(props?.selector ? { data } : { collection: data ?? null, ...result }),
  };
}
