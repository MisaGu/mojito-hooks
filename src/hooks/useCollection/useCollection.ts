import { QueryKey } from '../../domain/utils/queryKeyFactory.util';
import { config } from '../../domain/constants/general.constants';
import { EMojitoKey } from '../../domain/enums/state.enum';
import { CollectionBySlugResponse } from '../../domain/interfaces';
import {
  BaseQueryHookPropsWithUrlAndSlug,
  PaginatedQueryProps,
} from '../../domain/interfaces/hooks.interface';
import { collectionPreloadFn } from '../../domain/utils/service/collectionPreloadFn';
import { getCollectionSlugFromPathname } from '../../domain/utils/state/path.util';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';
import { getPaginationArgs } from '../../domain/utils/paginationUtils';

function selectorFn(response?: CollectionBySlugResponse) {
  if (!response) return undefined;

  return response.collectionBySlug;
}

export type UseCollectionData = ReturnType<typeof selectorFn>;

export type UseCollectionReturn = ReturnType<typeof useCollection>;

export interface UseCollectionProps
  extends BaseQueryHookPropsWithUrlAndSlug<UseCollectionData>,
    PaginatedQueryProps {
  marketplaceID: string;
}

export function useCollection(props: UseCollectionProps) {
  const collectionSlug = props.slug || getCollectionSlugFromPathname();
  const marketplaceID = props.marketplaceID ?? config.MARKETPLACE_ID;

  const paginationArgs = getPaginationArgs(props);

  return useMojitoFactory(
    {
      as: 'collection',
      queryKey: QueryKey.get(EMojitoKey.collectionBySlug, {
        slug: collectionSlug,
        marketplaceID,
        ...paginationArgs,
      }),
      options: props.options,
      onQueryStart: () => collectionPreloadFn(collectionSlug),
      selectorFn: selectorFn,
      force: true,
    },
    [collectionSlug, marketplaceID, paginationArgs?.offset, paginationArgs?.limit],
  );
}

export default useCollection;
