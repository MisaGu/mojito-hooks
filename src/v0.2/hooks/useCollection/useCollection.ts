import { useQueryClient } from 'react-query';
import { config } from '../../domain/constants/general.constants';
import {
  IContentfulAuctionsSlugListQuery,
  MojitoMarketplaceCollection,
  MarketplaceResponse,
  CollectionItemBySlugResponse,
} from '../../domain/interfaces';
import { getCollectionSlugFromPathname } from '../../domain/utils/state/path.util';
import { EMojitoKey } from '../../domain/enums/state.enum';
import { EContentfulKey } from '../../domain/enums/state.enum';
import { QueryKey } from '../../domain/utils/queryKeyFactory.util';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';
import { BaseQueryHookPropsWithUrlAndSlug } from '../../domain/interfaces/hooks.interface';

function selectorFn(response?: CollectionItemBySlugResponse) {
  if (!response) return undefined;

  return response.collectionBySlug;
}

export type UseCollectionData = ReturnType<typeof selectorFn>;

export type UseCollectionReturn = ReturnType<typeof useCollection>;

export type UseCollectionProps = BaseQueryHookPropsWithUrlAndSlug<UseCollectionData>;

export function useCollection(props?: UseCollectionProps) {
  const queryClient = useQueryClient();
  const collectionSlug = props?.slug || getCollectionSlugFromPathname();

  async function preloadFn() {
    const [mojitoCollections, contentfulCollectionSlugsOnly] = await Promise.all([
      queryClient.fetchQuery<MarketplaceResponse>(
        QueryKey.get(EMojitoKey.marketplaceCollectionsInfoWithItemsIdAndSlug, {
          id: config.MARKETPLACE_ID,
        }),
      ),
      queryClient.fetchQuery<IContentfulAuctionsSlugListQuery>(
        QueryKey.get(EContentfulKey.auctionsSlugList),
      ),
    ]);

    const mojitoCollection = mojitoCollections.marketplace.collections.find(
      (collection) => collection.slug == collectionSlug,
    );

    const contentfulCollection = contentfulCollectionSlugsOnly.auctionCollection.items.find(
      (collection) => collection.slug == collectionSlug,
    );

    if (!mojitoCollection) return;

    if (contentfulCollection) {
      const collectionItems = mojitoCollection.items.map((item) => item.id);

      await Promise.all([
        queryClient.prefetchQuery(
          QueryKey.get(EContentfulKey.auctionBySlug, { slug: collectionSlug }),
        ),
        queryClient.prefetchQuery(
          QueryKey.get(EContentfulKey.shortLots, { mojitoIds: collectionItems }),
        ),
      ]);
    }
  }

  return useMojitoFactory({
    as: 'collection',
    query: EMojitoKey.collectionBySlug,
    variables: {
      slug: collectionSlug,
      marketplaceID: config.MARKETPLACE_ID,
    },
    options: props?.options,
    preloadFn,
    selectorFn,
  });
}

export default useCollection;
