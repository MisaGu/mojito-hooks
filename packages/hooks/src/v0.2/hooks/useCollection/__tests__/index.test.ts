import { act, renderHook } from '@testing-library/react-hooks';
import { DEMO_SLUG } from '../../../components/demo/constants/demo.constants';
import { TestWrapper } from '../../../components/test/wrapper/TestWrapper';
import { useCollection, UseCollectionProps, UseCollectionReturn } from '../useCollection';

describe('useCollection()', () => {
  it('should be defined', () => {
    expect(useCollection).toBeDefined();
  });

  it('returns the right result when needed', async () => {
    const { result, rerender, waitFor } = renderHook<[UseCollectionProps], UseCollectionReturn>(
      (args) => {
        return useCollection(...args);
      },
      {
        initialProps: [{ slug: DEMO_SLUG }],
        wrapper: TestWrapper as any,
      },
    );

    // Trigger request...
    await act(async () => rerender());

    // ...and wait for the main query to start fetching...:
    await waitFor(() => result.current.queryResult.isFetching);

    // ...and for its response:
    await waitFor(() => result.current.queryResult.isSuccess);

    // Check data matches snapshot:
    expect(result.current).toMatchSnapshot({
      collection: {
        items: [
          {
            details: {
              endTimestamp: expect.any(Number),
            },
          },
        ],
      },
      queryResult: {
        dataUpdatedAt: expect.any(Number),
      },
    });
  });
});
