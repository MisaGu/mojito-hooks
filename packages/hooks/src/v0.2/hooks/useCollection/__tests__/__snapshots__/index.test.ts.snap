// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useCollection() returns the right result when needed 1`] = `
Object {
  "collection": Object {
    "contentfulData": undefined,
    "description": "",
    "endDate": "2022-05-25T23:36:00Z",
    "hasMultipleLots": false,
    "id": "mock-collection-id",
    "isFake": true,
    "items": Array [
      Object {
        "collectionId": "mock-collection-id",
        "contentfulData": Object {
          "lotId": -1,
          "mojitoId": "NA",
          "slug": "NA",
          "subtitle": "NA",
          "title": "NA",
        },
        "details": Object {
          "__typename": "MarketplaceBuyNowOutput",
          "currentBid": null,
          "endDate": "2022-05-25T23:36:00Z",
          "endTimestamp": Any<Number>,
          "id": "mock-collection-item-details-id",
          "saleView": Object {
            "isDuringSale": false,
            "isPostSale": true,
            "isPreSale": false,
          },
          "sortNumber": 0,
          "startDate": "2022-05-24T18:44:32Z",
          "totalAvailableUnits": 1,
          "totalUnits": 1,
          "unitPrice": 100,
        },
        "id": "mock-collection-item-id",
        "marketplaceTokenId": null,
        "name": "Mock Collection Item",
        "saleType": "BuyNow",
        "slug": "mock-collection-item",
      },
    ],
    "name": "Mock Collection",
    "slug": "mock-collection",
    "startDate": "2022-05-01T14:20:00Z",
    "viewStatus": Object {
      "hasActiveAuctionItems": false,
      "hasActiveBuyNowItems": false,
      "isDuringSale": false,
      "isPostSale": true,
      "isPreSale": false,
    },
  },
  "error": null,
  "isLoading": false,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": false,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
